Mage.Sets/src/mage/cards/r/RipplesOfPotential.java

```diff
@@ -0,0 +1,133 @@
+ */
+public class RipplesOfPotential extends CardImpl {
+
+    public RipplesOfPotential(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{1}{U}");
+
+        // Proliferate, then choose any number of permanents you control that had a counter put on them this way. Those permanents phase out.
+        // A watcher is required as another effect may prevent counters from being put on permanents (e.g. Solemnity)
+        this.getSpellAbility().addEffect(new ProliferateEffect());
+        this.getSpellAbility().addEffect(new RipplesOfPotentialEffect());
+        this.getSpellAbility().addWatcher(new RipplesOfPotentialWatcher());
+    }
+
+    private RipplesOfPotential(final RipplesOfPotential card) {
+        super(card);
+    }
+
+    @Override
+    public RipplesOfPotential copy() {
+        return new RipplesOfPotential(this);
+    }
+}
+
+class RipplesOfPotentialEffect extends OneShotEffect {
+
+    public RipplesOfPotentialEffect() {
+        super(Outcome.Benefit);
+        staticText = "Proliferate, then choose any number of permanents you control that had a counter put on them this way. Those permanents phase out.";
+    }
+
+    private RipplesOfPotentialEffect(final RipplesOfPotentialEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RipplesOfPotentialEffect copy() {
+        return new RipplesOfPotentialEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        RipplesOfPotentialWatcher watcher = game.getState().getWatcher(RipplesOfPotentialWatcher.class, source.getSourceId());
+        if (watcher != null) {
+            FilterPermanent filter = new FilterPermanent("permanents");
+            filter.add(Predicates.or(
+                    watcher
+                            .getProliferatedPermanents()
+                            .stream()
+                            .map(game::getPermanent)
+                            .filter(Objects::nonNull)
+                            .filter(permanent -> permanent.isControlledBy(source.getControllerId()))
+                            .map(Permanent::getId)
+                            .map(PermanentIdPredicate::new)
+                            .collect(Collectors.toSet())
+            ));
+            TargetPermanent target = new TargetPermanent(0, Integer.MAX_VALUE, filter, true);
+            controller.choose(outcome, target.withChooseHint("to phase out"), source, game);
+            for (UUID targetId : target.getTargets()) {
+                Permanent permanent = game.getPermanent(targetId);
+                if (permanent != null) {
+                    permanent.phaseOut(game);
+                }
+            }
+        }
+        return true;
+    }
+}
+
+class RipplesOfPotentialWatcher extends Watcher {
```

Comment:
Subtle bug in the watcher I think - if you manage to cast the same Ripples of Potential twice in one turn, you could phase out permanents that had a counter placed the first time but not the second. I think you need to clear the watcher's list after using it.